clients/src/main/java/org/apache/kafka/common/security/oauthbearer/OAuthBearerLoginModule.java

```diff
@@ -294,6 +331,18 @@ public boolean logout() {
             }
         }
         log.info("Done logging out my token; committed token count is now {}", committedTokenCount());
+
+        log.info("Logging out my extensions");
+        for (Iterator<SaslExtensions> iterator = subject.getPublicCredentials(SaslExtensions.class).iterator(); iterator.hasNext();) {
+            SaslExtensions extensions = iterator.next();
+            if (extensions.extensionsMap().equals(myCommittedExtensions)) {
```

Comment:
I think this will end up being `extensions == myCommitedExtensions` ?

Reply:
Nope - https://docs.oracle.com/javase/8/docs/api/java/util/AbstractMap.html#equals-java.lang.Object-

Reply:
Yeah, the use of == instead of .equals() is certainly unusual, but it is necessary in this case because we need to make sure we remove our instance as opposed to an instance put there via another LoginContext.  I did not comment it when I did it for the token but should have given that it is unusual.